pkg/northbound/admin/transaction.go

```
@@ -0,0 +1,84 @@
+	"github.com/onosproject/onos-config/pkg/store/transaction"
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+)
+
+// GetTransaction returns response with the requested transaction
+func (s Server) GetTransaction(ctx context.Context, req *admin.GetTransactionRequest) (*admin.GetTransactionResponse, error) {
+	log.Infof("Received GetTransaction request: %+v", req)
+	conf, err := s.transactionsStore.Get(ctx, req.ID)
+	if err != nil {
+		log.Warnf("GetTransaction %+v failed: %v", req, err)
+		return nil, errors.Status(err).Err()
+	}
+	return &admin.GetTransactionResponse{Transaction: conf}, nil
+}
+
+// ListTransactions provides stream listing all transactions
+func (s Server) ListTransactions(req *admin.ListTransactionsRequest, stream admin.TransactionService_ListTransactionsServer) error {
+	log.Infof("Received ListTransactions request: %+v", req)
+	transactions, err := s.transactionsStore.List(stream.Context())
+	if err != nil {
+		log.Warnf("ListTransactions %+v failed: %v", req, err)
+		return errors.Status(err).Err()
+	}
+	for _, conf := range transactions {
+		err := stream.Send(&admin.ListTransactionsResponse{Transaction: conf})
+		if err != nil {
+			log.Warnf("ListTransactions %+v failed: %v", req, err)
+			return errors.Status(err).Err()
+		}
+	}
+	return nil
+}
+
+// WatchTransactions provides stream with events representing transactions changes
+func (s Server) WatchTransactions(req *admin.WatchTransactionsRequest, stream admin.TransactionService_WatchTransactionsServer) error {
+	log.Infof("Received WatchTransactions request: %+v", req)
+	var watchOpts []transaction.WatchOption
+	if !req.Noreplay {
+		watchOpts = append(watchOpts, transaction.WithReplay())
+	}
+
+	ch := make(chan configapi.TransactionEvent)
+	if err := s.transactionsStore.Watch(stream.Context(), ch, watchOpts...); err != nil {
+		log.Warnf("WatchTransactionsRequest %+v failed: %v", req, err)
+		return errors.Status(err).Err()
+	}
+
+	return s.streamTransactions(stream, ch)
```

I think we should return the error here and convert it to status before returning it back.

---

Fixed